pulse2percept/oyster.py

```diff
@@ -41,149 +42,161 @@ def jansonius(nCells, nR, center=np.array([15,2]),
         axon_xg = axon_xg + ([cur_xg],)
         axon_yg = axon_yg + ([cur_yg],)
         axon_id = axon_id + ([id],)
 
-        #now loop back along this nearest axon toward the optic disc
-        for axPosId in range(axPosId0-1, -1, -1):
-            #increment the distance from the starting point
-            dist = dist + np.sqrt((axon_x[axPosId+1,axNum]-axon_x[axPosId,axNum])**2
-                    + (axon_y[axPosId+1,axNum]-axon_y[axPosId,axNum])**2)
-
-            weight = np.exp(-dist/axon_lambda)  # weight falls off exponentially as distance from axon cell body
-
-            #find the nearest pixel to the current position along the axon
-            nearest_xg_id = np.abs(xg[0,:]-axon_x[axPosId,axNum]).argmin()
-            nearest_yg_id = np.abs(yg[:,0]-axon_y[axPosId,axNum]).argmin()
-            nearest_xg =xg[0,nearest_xg_id]
-            nearest_yg =yg[nearest_yg_id,0]
-
-            #if the position along the axon has moved to a new pixel, and the weight isn't too small...
-            if nearest_xg != cur_xg or nearest_yg != cur_yg and weight>min_weight:
-                #update the current pixel location
-                cur_xg = nearest_xg
-                cur_yg = nearest_yg
-
-                #append the list
-                #axon_dist[id].append(dist)
-                axon_weight[id].append(np.exp(-dist/axon_lambda))
-                #axon_xg[id].append(cur_xg)
-                #axon_yg[id].append(cur_yg)
-                axon_id[id].append(np.ravel_multi_index((nearest_yg_id,
-                                   nearest_xg_id),xg.shape))
+        # now loop back along this nearest axon toward the optic disc
+        for axPosId in range(axPosId0 - 1, -1, -1):
+            # increment the distance from the starting point
+            ax = (axon_x[axPosId + 1, axNum] - axon_x[axPosId, axNum])**2
+            ay = (axon_y[axPosId + 1, axNum] - axon_y[axPosId, axNum])**2
+            dist += np.sqrt(ax ** 2 + ay ** 2)
+
+            # weight falls off exponentially as distance from axon cell body
+            weight = np.exp(-dist / axon_lambda)
+
+            # find the nearest pixel to the current position along the axon
+            nearest_xg_id = np.abs(xg[0, :] - axon_x[axPosId, axNum]).argmin()
+            nearest_yg_id = np.abs(yg[:, 0] - axon_y[axPosId, axNum]).argmin()
+            nearest_xg = xg[0, nearest_xg_id]
+            nearest_yg = yg[nearest_yg_id, 0]
+
+            # if the position along the axon has moved to a new pixel, and the
+            # weight isn't too small...
+            if weight > min_weight:
+                if nearest_xg != cur_xg or nearest_yg != cur_yg:
+                    # update the current pixel location
+                    cur_xg = nearest_xg
+                    cur_yg = nearest_yg
+
+                    # append the list
+                    # axon_dist[id].append(dist)
+                    axon_weight[id].append(np.exp(-dist / axon_lambda))
```

Comment:
Good point, I'll replace it.

The bug was in `if nearest_xg != cur_xg or nearest_yg != cur_yg and weight>min_weight`. The or condition should be grouped together: `(if nearest_xg != cur_xg or nearest_yg != cur_yg) and weight>min_weight`. I nested the conditions to make it clearer (and fit on a PEP8-compliant line).